ts-sdk/whirlpool/tests/createPool.test.ts

```diff
@@ -1,5 +1,263 @@
+    await assert.rejects(
+      createConcentratedLiquidityPoolInstructions(rpc, mintB, mintA, 64, 1),
+    );
+  });
+
+  it("Should create splash pool", async () => {
+    const price = 10;
+    const sqrtPrice = priceToSqrtPrice(price, 6, 6);
+
+    let signerAccount = await rpc.getAccountInfo(signer.address).send();
+    const balanceBefore = signerAccount.value?.lamports ?? lamports(0n);
+
+    const { instructions, poolAddress, estInitializationCost } =
+      await createSplashPoolInstructions(rpc, mintA, mintB, price);
+
+    const maybePool = await fetchMaybeWhirlpool(rpc, poolAddress);
+    assert.strictEqual(maybePool.exists, false);
+
+    await sendTransaction(instructions);
+
+    const pool = await fetchMaybeWhirlpool(rpc, poolAddress);
+    assertAccountExists(pool);
+
+    signerAccount = await rpc.getAccountInfo(signer.address).send();
+    const balanceAfter = signerAccount.value?.lamports ?? lamports(0n);
+    const balanceChange = balanceBefore - balanceAfter;
+    const txFee = 15000n; // 3 signing accounts * 5000 lamports
+    const minRentExempt = balanceChange - txFee;
+    assert.strictEqual(estInitializationCost, minRentExempt);
+
+    assert.strictEqual(sqrtPrice, pool.data.sqrtPrice);
+    assert.strictEqual(mintA, pool.data.tokenMintA);
+    assert.strictEqual(mintB, pool.data.tokenMintB);
+    assert.strictEqual(SPLASH_POOL_TICK_SPACING, pool.data.tickSpacing);
+  });
+
+  it("Should create splash pool with 1 TE token", async () => {
+    const mint1 = await setupMint();
+    const mint2 = await setupMintTEFee();
+    const [mintC, mintD] = orderMints(mint1, mint2);
+    const price = 10;
+    const sqrtPrice = priceToSqrtPrice(price, 6, 6);
+
+    let signerAccount = await rpc.getAccountInfo(signer.address).send();
+    const balanceBefore = signerAccount.value?.lamports ?? lamports(0n);
+
+    const { instructions, poolAddress, estInitializationCost } =
+      await createSplashPoolInstructions(rpc, mintC, mintD, price);
+
+    const maybePool = await fetchMaybeWhirlpool(rpc, poolAddress);
+    assert.strictEqual(maybePool.exists, false);
+
+    await sendTransaction(instructions);
+
+    const pool = await fetchMaybeWhirlpool(rpc, poolAddress);
+    assertAccountExists(pool);
+
+    signerAccount = await rpc.getAccountInfo(signer.address).send();
+    const balanceAfter = signerAccount.value?.lamports ?? lamports(0n);
+    const balanceChange = balanceBefore - balanceAfter;
+    const txFee = 15000n; // 3 signing accounts * 5000 lamports
+    const minRentExempt = balanceChange - txFee;
+    assert.strictEqual(estInitializationCost, minRentExempt);
+
+    assert.strictEqual(sqrtPrice, pool.data.sqrtPrice);
+    assert.strictEqual(mintC, pool.data.tokenMintA);
+    assert.strictEqual(mintD, pool.data.tokenMintB);
+    assert.strictEqual(SPLASH_POOL_TICK_SPACING, pool.data.tickSpacing);
+  });
+
+  it("Should create splash pool with 2 TE tokens", async () => {
+    const mint1 = await setupMintTEFee();
```

Comment:
For the tests I would do (instead of trying to combine Transfer fee and 1/2 TE token tests):
* 1 TE token (without extensions)
* 2 TE tokens (without extensions)
* 1 TE token (with transfer fee)